### PR TITLE
Implements shadow mode backend

### PR DESCRIPTION
**Pull request by Tidyzq**
related #1

### CODE REVIEW

File: glass-easel/src/backend/composed_backend_protocol.ts [hunk 1]
```diff
@@ -30,21 +23,17 @@ export interface Context extends Partial<suggestedBackend.Context> {
   createFragment(): Element
   onEvent(
     listener: (
-      target: unknown,
+      target: GlassEaselElement,
       type: string,
       detail: unknown,
       options: EventOptions,
-    ) => EventBubbleStatus,
+    ) => EventBubbleStatus | void,
```
Comment (LastLeaf) [inline glass-easel/src/backend/composed_backend_protocol.ts:30]: Why?

File: glass-easel/src/backend/composed_backend_protocol.ts [hunk 2]
```diff
@@ -30,21 +23,17 @@ export interface Context extends Partial<suggestedBackend.Context> {
   createFragment(): Element
   onEvent(
     listener: (
-      target: unknown,
+      target: GlassEaselElement,
```
Comment (LastLeaf) [inline glass-easel/src/backend/composed_backend_protocol.ts:26]: If shared a backend context with another glass-easel version or other frameworks, the event target may not be an `Element` of current glass-easel. Thus this should be unknown and requires an extra type check.

File: glass-easel/src/backend/composed_backend_protocol.ts (hunk 1 above)
Comment (Tidyzq) [reply glass-easel/src/backend/composed_backend_protocol.ts:30]: > Why?

Developers may simply not returning anything

File: glass-easel/src/global_options.ts [hunk 3]
```diff
@@ -191,7 +191,7 @@ export const normalizeComponentOptions = (
     templateEngine: p.templateEngine !== undefined ? p.templateEngine : b.templateEngine,
     styleScope: p.styleScope !== undefined ? p.styleScope : b.styleScope,
     hostNodeTagName: p.hostNodeTagName !== undefined ? p.hostNodeTagName : b.hostNodeTagName,
-    extraStyleScope: p.extraStyleScope !== undefined ? p.extraStyleScope : b.extraStyleScope,
+    extraStyleScope: p.extraStyleScope ?? b.extraStyleScope,
```
Comment (LastLeaf) [inline glass-easel/src/global_options.ts:194]: Update with `extraStyleScope: null` will fail!

File: glass-easel/src/backend/composed_backend_protocol.ts (hunk 2 above)
Comment (Tidyzq) [reply glass-easel/src/backend/composed_backend_protocol.ts:26]: > If shared a backend context with another glass-easel version or other frameworks, the event target may not be an `Element` of current glass-easel. Thus this should be unknown and requires an extra type check.

This type is to constraint backend's behavior, backends should pass a `glassEasel.Element` into this callback。

We can add an aditional type check inside this callback to solve the multi-version problem, but still need this type to tell backend developers that we are expecting a `glassEasel.Element` here.

**Pull request merged**